### PR TITLE
Implement Player and Martingale class

### PR DESCRIPTION
**Pull request by FarzanRashid**
This PR includes Player and Martingale classes for the roulette game. This PR also has tests for both classes.

### CODE REVIEW

File: src/roulette.py [hunk 1]
```diff
@@ -676,7 +677,53 @@ def __repr__(self) -> str:
...
+        pass
+
+    @abstractmethod
+    def placeBets(self) -> None:
+        pass  # pragma: no cover
+
+    def playing(self) -> bool:
+        if self.stake < self.table.minimum:
+            return False
+        return True
```
Comment (FarsanRashid) [inline src/roulette.py:699]: can be simplified using a single return statement

File: tests/test_game.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 from unittest.mock import Mock, patch
 
+import roulette
```
Comment (FarsanRashid) [inline tests/test_game.py:5]: unused import.

File: tests/test_game.py [hunk 3]
```diff
@@ -47,7 +48,18 @@ def test_cycle_calls_lose_if_bet_loses(self):
     def test_choose_not_called_if_isValid_raises_exception(self):
         is_valid_mock = Mock(name="is_valid_mock", side_effect=InvalidBet)
         choose_mock = Mock(name="choose_mock")
-        with patch("roulette.Table.isValid", is_valid_mock):
+        with patch.multiple("roulette.Table", isValid=is_valid_mock), patch.multiple(
+            "roulette.Wheel", choose=choose_mock
+        ):
```
Comment (FarsanRashid) [inline tests/test_game.py:53]: `patch.multiple` is better suited when patching multiple functions of the same class

File: tests/test_game.py [hunk 4]
```diff
@@ -47,7 +48,18 @@ def test_cycle_calls_lose_if_bet_loses(self):
...
             with self.assertRaises(InvalidBet):
                 self.game.cycle(self.passenger)
         choose_mock.assert_not_called()
+
+    def test_cycle_not_executed_if_no_active_players(self):
+        playing_mock = Mock(name="playing_mock", return_value=False)
+        place_bets_mock = Mock(name="place_bets_mock")
+        with patch.multiple("roulette.Player", playing=playing_mock), patch.multiple(
+            "roulette.Player", placeBets=place_bets_mock
+        ):
```
Comment (FarsanRashid) [inline tests/test_game.py:63]: single use of `patch.multiple` can patch multiple attributes

File: tests/test_martingale.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+class TestMartingale(TestCase):
+    def setUp(self):
+        self.table = Table()
+        self.wheel = Wheel()
+        bin_builder = BinBuilder()
+        bin_builder.buildBins(self.wheel)
+        self.game = Game(self.wheel, self.table)
+        self.martingale = Martingale(self.table)
+
+    def test_placeBets_make_bets_if_enough_stake(self):
```
Comment (FarsanRashid) [inline tests/test_martingale.py:17]: we don't test functions rather behavior. so a better name can be `test_placing_bets_need_minimum_stake`.

Comment (FarsanRashid) [reply tests/test_martingale.py:17]: This test can cover both cases (enough stake/less stake)

Comment (FarzanRashid) [reply tests/test_martingale.py]: Done.

File: tests/test_game.py (hunk 2 above)
Comment (FarzanRashid) [reply tests/test_game.py]: Removed.

File: tests/test_martingale.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+class TestMartingale(TestCase):
+    def setUp(self):
+        self.table = Table()
+        self.wheel = Wheel()
+        bin_builder = BinBuilder()
+        bin_builder.buildBins(self.wheel)
+        self.game = Game(self.wheel, self.table)
+        self.martingale = Martingale(self.table)
+
+    def test_placing_bets_need_minimum_stake(self):
```
Comment (FarsanRashid) [inline tests/test_martingale.py:17]: better rename to `test_bets_placed_when_stake_not_less_than_betMultiple`

File: tests/test_game.py [hunk 7]
```diff
@@ -48,6 +48,16 @@ def test_choose_not_called_if_isValid_raises_exception(self):
...
         choose_mock = Mock(name="choose_mock")
         with patch("roulette.Table.isValid", is_valid_mock):
-            with self.assertRaises(InvalidBet):
-                self.game.cycle(self.passenger)
+            with patch("roulette.Wheel.choose", choose_mock):
+                with self.assertRaises(InvalidBet):
+                    self.game.cycle(self.passenger)
         choose_mock.assert_not_called()
+
+    def test_cycle_not_executed_if_no_active_players(self):
```
Comment (FarsanRashid) [inline tests/test_game.py:56]: better rename to `test_bets_not_placed_for_inactive_player` as cycle is being executed but bets are not placed when player is not playing

Comment (FarsanRashid) [reply tests/test_game.py:56]: It is obvious that `placeBets` will not be called if player is not playing. Adding test against it is not much helpful. so can be removed.

File: tests/test_martingale.py [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    def test_stake_raised_if_bet_wins(self):
+        self.martingale.betMultiple = 10
+        with patch(
+            "roulette.Wheel.choose", Mock(return_value=[self.wheel.getOutcome("Black")])
+        ):
+            self.game.cycle(self.martingale)
+        expected_stake = 110
+        self.assertEqual(expected_stake, self.martingale.stake)
+
+    def test_playing_returns_false_if_stake_less_than_limit(self):
```
Comment (FarsanRashid) [inline tests/test_martingale.py:47]: better rename to `test_player_plays_when_stake_not_less_than_limit` and test both `True` and `False` scenario.

File: tests/test_martingale.py [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+        self.martingale.stake = 3
+        playing_result = self.martingale.playing()
+        self.assertEqual(False, playing_result)
```
Comment (FarsanRashid) [inline tests/test_martingale.py:50]: better use `self.assertFalse`

File: tests/test_martingale.py (hunk 6 above)
Comment (FarzanRashid) [reply tests/test_martingale.py]: Done.

File: tests/test_game.py (hunk 7 above)
Comment (FarzanRashid) [reply tests/test_game.py]: Done.

File: tests/test_martingale.py (hunk 9 above)
Comment (FarzanRashid) [reply tests/test_martingale.py:50]: Done.

File: tests/test_martingale.py (hunk 8 above)
Comment (FarzanRashid) [reply tests/test_martingale.py]: Done.

File: tests/test_martingale.py [hunk 10]
```diff
@@ -47,4 +47,8 @@ def test_stake_raised_if_bet_wins(self):
     def test_player_plays_when_stake_not_less_than_table_minimum(self):
         self.martingale.stake = 3
         playing_result = self.martingale.playing()
-        self.assertEqual(False, playing_result)
+        self.assertFalse(False)
```
Comment (FarsanRashid) [inline tests/test_martingale.py:50]: Should check `playing_result`. Better invoke `playing` in place

Comment (FarzanRashid) [reply tests/test_martingale.py]: Done.

**Pull request merged**